segfast/segyio_loader.py

```diff
@@ -125,32 +123,60 @@ def load_headers(self, headers, reconstruct_tsf=True, sort_columns=True, tracewi
             If str, then type of progress bar to display: `'t'` for textual, `'n'` for widget.
         """
         _ = kwargs
-        headers = list(headers)
-
-        if reconstruct_tsf and 'TRACE_SEQUENCE_FILE' in headers:
-            headers.remove('TRACE_SEQUENCE_FILE')
+        headers = self._make_headers_specs(headers)
+        if not all([header.is_standard_except_name for header in headers]):
+            warnings.warn("All nonstandard trace headers byte positions and dtypes will be ignored.")
 
-        headers_bytes = self.headers_to_bytes(headers)
+        if reconstruct_tsf:
+            headers = [header for header in headers if header.name != 'TRACE_SEQUENCE_FILE']
 
         # Load data to buffer
         buffer = np.empty((self.n_traces, len(headers)), dtype=np.int32)
         if tracewise:
             for i, header_ in Notifier(pbar, total=self.n_traces, frequency=1000)(enumerate(self.file_handler.header)):
-                for j, byte_ in enumerate(headers_bytes):
-                    buffer[i, j] = header_.getfield(header_.buf, byte_)
+                for j, header in enumerate(headers):
+                    buffer[i, j] = header_.getfield(header_.buf, header.start_byte)
         else:
             for i, header in enumerate(headers):
                 buffer[:, i] = self.load_header(header)
 
         # Convert to pd.DataFrame, optionally add TSF and sort
-        dataframe = pd.DataFrame(buffer, columns=headers, copy=False)
+        dataframe = pd.DataFrame(buffer, columns=[item.name for item in headers], copy=False)
         dataframe = self.postprocess_headers_dataframe(dataframe, headers=headers,
                                                        reconstruct_tsf=reconstruct_tsf, sort_columns=sort_columns)
         return dataframe
 
+    def _make_headers_specs(self, headers):
```

Comment:
It would be very useful for us to have this as a public function (without using loader byteorder, obviously)